### PR TITLE
Removed file read during handle_config to prevent hang at boot.

### PR DESCRIPTION
**Pull request by qharley**
Related to consecutive fopen on SD.

### CODE REVIEW

**Comment by qharley** [comment]
One of my machines would sporadically hang on boot - found that consecutive fopen needs a delay to prevent problems.

**Comment by wolfmanjm** [comment]
ping me on irc dev channel we can discuss how to do this. I have an idea how to do it.

**Comment by qharley** [comment]
You can now save a grid with:
M374 MYCUSTOMGRID
and it will save it in "/sd/Zgrid.MYCUSTOMGRID"

**Comment by qharley** [comment]
I will test it on a machine later today

File: src/modules/tools/zprobe/ZGridStrategy.cpp [hunk 1]
```diff
@@ -251,20 +247,30 @@ bool ZGridStrategy::handleGcode(Gcode *gcode)
 
             // M374: Save grid
             case 374:{
-                if(this->saveGrid()) {
+               char gridname[5];
+
+               if(gcode->has_letter('S'))  // Custom grid number
+                    sprintf(gridname, "S%3.0f", gcode->get_value('S'));
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: This is too dangerous you will get stack corruption and crash if someone enter M374 S123456
You must use snprintf(gridname, 4, ...)

File: src/modules/tools/zprobe/ZGridStrategy.cpp [hunk 2]
```diff
@@ -251,20 +247,30 @@ bool ZGridStrategy::handleGcode(Gcode *gcode)
...
             }
             return true;
 
-            case 375: // Load grid values
-                if(this->loadGrid()){
+            case 375:{ // Load grid values
+                char gridname[5];
+
+                if(gcode->has_letter('S'))  // Custom grid number
+                    sprintf(gridname, "S%3.0f", gcode->get_value('S'));
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: ditto here use snprintf()

File: src/modules/tools/zprobe/ZGridStrategy.cpp [hunk 3]
```diff
@@ -288,6 +294,8 @@ bool ZGridStrategy::handleGcode(Gcode *gcode)
             return true;
 
             case 500: // M500 saves probe_offsets config override file
+                gcode->stream->printf("M375  ; Load default grid\n");
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: only if the config option save grid is set to true. as not everyone will want the grid loaded on startup.

File: src/modules/tools/zprobe/ZGridStrategy.cpp [hunk 4]
```diff
@@ -327,24 +332,19 @@ bool ZGridStrategy::saveGrid() //std::string *args)
 
 }
 
-bool ZGridStrategy::loadGrid()
+bool ZGridStrategy::loadGrid(std::string args)
```
Comment (wolfmanjm) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: inefficient.. should be const std::string& args, or better still const char* args)

Comment (qharley) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: It may be inefficient, but the pointers are messing with my head today.

At least this works. ;-)

Comment (minad) [inline src/modules/tools/zprobe/ZGridStrategy.cpp]: @wolfmanjm I was already wondering about references in the current codebase which is not consistent in using those where it makes sense.

**Comment by wolfmanjm** [comment]
FYI a bug is here https://github.com/RepRapMorgan/Smoothieware/blob/zgrid/src/modules/tools/zprobe/ZGridStrategy.cpp#L144

The pData is never initialized to nullptr so can crash randomly here. someone already ran into this bug.

**Pull request merged**